src/main/java/org/springframework/data/elasticsearch/core/DefaultResultMapper.java

```diff
@@ -206,4 +209,16 @@ private String buildJSONFromFields(Collection<DocumentField> values) {
 			}
 		}
 	}
+
+	private <T> void setPersistentEntityScore(T result, float score, Class<T> clazz) {
+		if (mappingContext != null && clazz.isAnnotationPresent(Document.class)) {
+			ElasticsearchPersistentEntity<?> persistentEntity = mappingContext.getRequiredPersistentEntity(clazz);
+			ElasticsearchPersistentProperty scoreProperty = persistentEntity.getScoreProperty();
+			Class<?> type = scoreProperty.getType();
```

Comment:
Does it make sense to already reject score properties with other types than `float`/`Float` when the `ElasticsearchPersistentProperty` is created? That would fail much faster and not let the application bootstrap and the score not being mapped without any indication that the type doesn't match.